router/router.go

```diff
@@ -123,20 +124,34 @@ func New(cfg Config) http.Handler {
 
 	if cfg.RelCalEnabled {
 		if cfg.UseNewReleaseCalendar {
-			router.Handle(cfg.RelCalRoutePrefix+"/releasecalendar", relcal.Handler(cfg.RelCalHandler))
-			router.Handle(cfg.RelCalRoutePrefix+"/releases/{uri:.*}", relcal.Handler(cfg.RelCalHandler))
+			if appConfig.LegacyCacheProxyEnabled {
+				router.Handle(cfg.RelCalRoutePrefix+"/releasecalendar", relcal.Handler(cfg.RelCalHandler))
+				router.Handle(cfg.RelCalRoutePrefix+"/releases/{uri:.*}", relcal.Handler(cfg.ProxyHandler))
+			} else {
+				router.Handle(cfg.RelCalRoutePrefix+"/releasecalendar", relcal.Handler(cfg.RelCalHandler))
+				router.Handle(cfg.RelCalRoutePrefix+"/releases/{uri:.*}", relcal.Handler(cfg.RelCalHandler))
+			}
 		} else {
-			router.Handle(cfg.RelCalRoutePrefix+"/releasecalendar", relcal.Handler(cfg.BabbageHandler))
-			router.Handle(cfg.RelCalRoutePrefix+"/releases/{uri:.*}", relcal.Handler(cfg.BabbageHandler))
+			if appConfig.LegacyCacheProxyEnabled {
+				router.Handle(cfg.RelCalRoutePrefix+"/releasecalendar", relcal.Handler(cfg.BabbageHandler))
+				router.Handle(cfg.RelCalRoutePrefix+"/releases/{uri:.*}", relcal.Handler(cfg.ProxyHandler))
+			} else {
+				router.Handle(cfg.RelCalRoutePrefix+"/releasecalendar", relcal.Handler(cfg.BabbageHandler))
+				router.Handle(cfg.RelCalRoutePrefix+"/releases/{uri:.*}", relcal.Handler(cfg.BabbageHandler))
+			}
 		}
 		router.Handle(cfg.RelCalRoutePrefix+"/calendar/releasecalendar", cfg.RelCalHandler)
 	}
 
-	// if the request is for a file go directly to babbage instead of using the allRoutesMiddleware
-	router.MatcherFunc(hasFileExtMatcher).Handler(cfg.BabbageHandler)
-
-	// If it is a known babbage endpoint go directly to babbage instead of using the allRoutesMiddleware
-	router.MatcherFunc(isKnownBabbageEndpointMatcher).Handler(cfg.BabbageHandler)
+	if appConfig.LegacyCacheProxyEnabled {
+		router.MatcherFunc(hasFileExtMatcher).Handler(cfg.ProxyHandler)
+		router.MatcherFunc(isKnownBabbageEndpointMatcher).Handler(cfg.ProxyHandler)
+	} else {
```

Comment:
Both blocks of the if/else are identical, except for the handler that's passed to the `Handler` function. That means that you should probably revert these changes and make the handler a variable (if proxy enabled, then handler=proxyHandler, else handler=babbageHandler).